### PR TITLE
Create department and faculty

### PR DESCRIPTION
**Pull request by Timbaron**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,4 @@ Homestead.json
 Homestead.yaml
 npm-debug.log
 yarn-error.log
+test.http
```
Comment (otilor) [inline .gitignore:14]: I don't think I need this file. Do I?🤔

File: app/Http/Controllers/API/StudentController.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?php
```
Comment (otilor) [inline app/Http/Controllers/API/StudentController.php:1]: What is the essence of this file?

File: app/Models/User.php [hunk 3]
```diff
@@ -16,11 +16,7 @@ class User extends Authenticatable
      *
      * @var array
      */
-    protected $fillable = [
-        'name',
-        'email',
-        'password',
-    ];
+    protected $gaurded = [];
```
Comment (otilor) [inline app/Models/User.php]: First off, I know guarded provides you an easy way of mass assigning but then you might have issues when trying to update some guarded properties later on. You might result to having forceFills in your codebase which is not the best idea.

A Hacker can also mass update some columns that are not guarded in your database say ```is_admin``` column which is not supposed to be.

File: database/migrations/2021_04_07_101239_add_more_columns_to_users_table.php [hunk 4]
```diff
@@ -16,6 +16,7 @@ public function up()
         Schema::table('users', function (Blueprint $table) {
             $table->string('matric_no')->nullable()->after('email');
             $table->date('date_of_birth')->after('matric_no');
+
```
Comment (otilor) [inline database/migrations/2021_04_07_101239_add_more_columns_to_users_table.php:19]: ???

File: database/migrations/2021_04_07_211228_create_departments_table.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('departments', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->unsignedBigInteger('faculty_id')->nullable();
+            $table->foreign('faculty_id')->references('id')->on('faculties');
```
Comment (otilor) [inline database/migrations/2021_04_07_211228_create_departments_table.php:20]: This is not necessary. You have already defined the relationship in your model class

File: database/migrations/2021_04_07_212814_add_department_id_to_users_table.php [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('users', function (Blueprint $table) {
+            $table->unsignedBigInteger('department_id')->after('date_of_birth');
+            $table->foreign('department_id')->references('id')->on('departments');
```
Comment (otilor) [inline database/migrations/2021_04_07_212814_add_department_id_to_users_table.php:18]: This also is not necessary.

File: database/seeders/DatabaseSeeder.php [hunk 7]
```diff
@@ -13,6 +14,11 @@ class DatabaseSeeder extends Seeder
      */
     public function run()
     {
-        // \App\Models\User::factory(10)->create();
+        $this->call([
+            FacultySeeder::class,
+            DepartmentSeeder::class,
+        ]);
+        
+        User::factory(10)->create();
```
Comment (otilor) [inline database/seeders/DatabaseSeeder.php]: You can just write this in UserSeeder then call it just like you did for the others.

File: database/seeders/DepartmentSeeder.php [hunk 8]
```diff
@@ -0,0 +1,28 @@
+<?php
```
Comment (otilor) [inline database/seeders/DepartmentSeeder.php:1]: This seeder assigns any faculty to a department which is not supposed to be.

File: database/seeders/FacultySeeder.php [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+class FacultySeeder extends Seeder
+{
+    /**
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+        $faculties = ['Faculty of Humanities','Faculty of Social and Management Science','Faculty of Science and Science Education'];
```
Comment (otilor) [inline database/seeders/FacultySeeder.php]: You don't need to put "Faculty of ". Just call it "Humanities", "Social and Management Sciences", "Science and Science Education"

File: routes/api.php [hunk 10]
```diff
@@ -23,6 +23,7 @@
     return response()->json(['message' => 'Welcome to Studently API', 'status' => 200]);
 });
 
+
```
Comment (otilor) [inline routes/api.php:28]: ???

**Review by otilor — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Timbaron) [reply .gitignore:14]: You don't. Unless you want to test the API in your IDE using Rest Client.

File: app/Models/User.php (hunk 3 above)
Comment (Timbaron) [reply app/Models/User.php]: Fixed

File: database/seeders/DatabaseSeeder.php [hunk 11]
```diff
@@ -13,6 +14,7 @@ class DatabaseSeeder extends Seeder
      */
     public function run()
     {
+        Artisan::call('migrate:fresh');
```
Comment (otilor) [inline database/seeders/DatabaseSeeder.php:17]: We don't need to do this.

**Review by otilor — APPROVED** [review]

**Pull request merged**